### ISSUE TITLE
Ignore or record last valid data when outdoor module disconnected

### ISSUE THREAD

**Issue by andrewtzn** [skrollme/homebridge-eveatmo#34]
Hi, 

First of all, I would like to thank you for creating and maintaining this plugin.

I would like to report a bug.

Whenever outdoor module connection as lost the data will get recorded as 0 on temperature and humidity. 
Is it possible to have the plugin ignore the invalid data? or it can just record the latest valid data until it can receive new data.

Here are some screenshot of the issue:
<img src="https://user-images.githubusercontent.com/9326712/69188020-b7786580-0ae9-11ea-9881-4a14c53009f0.PNG" width="300" height="600">
<img src="https://user-images.githubusercontent.com/9326712/69188028-b9dabf80-0ae9-11ea-8618-3f9b94d591d5.PNG" width="300" height="600">

I have the latest plugin v0.4.8 running on Homebirdge 0.4.50. 

Thank you!

**Comment by skrollme** [comment]
Hi @andrewtzn , I will definetly take a look at this, but be aware that this is just a spare-time project so it may take some time to dig deeper into this.

If you are "a little bit into nodeJS", feel free to start going through the code and make suggestions or even create a PullRequest ;)

**Comment by andrewtzn** [comment]
Thank you for your reply @skrollme !
Please take your time. I'm very grateful to have you work on this.

For sure, I'll take a look at it. I'll post feedback if I find anything.
Do you know where is the temperature and humidity cache data stored on a Homebridge server?

**Comment by skrollme** [comment]
> Do you know where is the temperature and humidity cache data stored on a Homebridge server?

On my setup the persistence-files from the history-plugin inside homebridge-eveatmo are located inside the _~/.homebridge_ directory. There should be one file per device and - at least in my case - they are named after the host and the netatmo-devicename:

```
raspberrypi_Zuhause Außen_persist.json
raspberrypi_Zuhause Schlafzimmer_persist.json
raspberrypi_Zuhause Wohnzimmer_persist.json
```

I took a short look at the Netatmo API to look how to detect a "non talking module" and I have a clue, but I think it would be better to test this by removing the batteries from my outdoor module and look what happens. I will let you know if this continues...

**Comment by andrewtzn** [comment]
Thank you @skrollme ! I'll take a look at those files

Approaching the issue from API make more sense than approaching from the fetched data.

Thank you again for your efforts!

**Comment by skrollme** [comment]
Hello @andrewtzn 
please take a look at the just released version 0.4.9 and let me know if this works for you.

**Comment by andrewtzn** [comment]
Hi @skrollme ,

Thank you for the update. I appreciated it!

However, I think there is an issue. The update seems to make my Homebridge unreachable.
When I installed the plugin, all of my accessories were showing "No Response" on the Home app.
When I removed plugin and the configs for this plugin on the config file, everything is reachable again.

I even started from stretch and tested to see if the update caused the issue.
I think the update is causing some issue.

Some info on my setup:
homebridge-config-ui-x v4.6.5 - node.js v10.16.0 

Thanks again!

**Comment by skrollme** [comment]
Interesting, I'm also using the new version since release (raspberryPI, homebride in Docker) and I have no problems at all.
Do you have some more details, error-messages from the log, or something else?

**Comment by andrewtzn** [comment]
I did checked the log on the Homebridge server, but there were no errors. 
Will I be able to find more detailed logs on the plugin level?

However, I think I found out what the issue was.
The outdoor module was failing and wasn't sending any data yesterday and Friday. 
I retrieved the outdoor module and it was able to turn on after one night.

I re-added the plugin and pasted the configs, and there is no issue at all!

I suspect if the plugin didn't get data initially the issue will occur?

Thank you!

**Comment by skrollme** [comment]
Hey @andrewtzn , thank you for your help. It could indeed be a problem, if a sensor is not available while homebridge boots up. I will take a look at this

**Comment by andrewtzn** [comment]
Hi @skrollme !
No problem! I'm glad to be part of the debugging process. :)

**Comment by skrollme** [comment]
Hi @andrewtzn ,
I just published an update which could fix the startup problem in 0.4.9. Nevertheless, the plugin was not designed to start up with missing / not-responding modules. I hope it works 🙈

🎄Happy holidays 🎄

**Comment by andrewtzn** [comment]
Thank you @skrollme !
I really appreciated your continuous support! 

I will close the issue now.

Same to you! Happy holidays!
Happy new year! 🎉
